transmogrifier/sources/transformer.py

```diff
@@ -191,3 +193,39 @@ def get_valid_title(cls, source_record_id: str, xml: Tag) -> str:
             )
             title = "Title not provided"
         return title
+
+    @classmethod
+    def get_source_link(
+        cls, source_base_url: str, source_record_id: str, xml: Tag
```

Comment:
Do we need `xml` as a parameter here if it's not being used? Couldn't the method just be overridden in the source transforms with an `xml` parameter when it's needed?

Reply:
Nice catch, and I had also paused on this.  I think if these are getting decoupled with the understanding transformers may override them, it's plausible they might need the document XML, so I included here.

Because the base transform calls this method, it will always use the same signature, so a sub-class transformer couldn't add `xml:Tag` on their own.  

We could adopt a `kwargs` approach, where the sub-class transformer _could_ pass anything they want as named arguments, but we'd lose the typing this provides.  Other than `xml:Tag`, it's hard to guess what they may want to pass?  

See pros and cons both ways.  My thinking was to model these after other methods that are designed to overridden, like `get_valid_title()` or `get_source_record_id()` which both pass `xml:Tag`.

Reply:
But if helpful, we could add the docstring comment:

```
xml: A BeautifulSoup Tag representing a single XML record.
                - not used by default implementation, but could be useful for subclass
                overrides
```

which `get_timdex_record_id()` contains below.  I had actually intended to do so.

Reply:
Let's just add the comment, I did some more reading and see your point on losing the type hinting